### ISSUE TITLE
Failure on export of reports with large number of attachments

### ISSUE THREAD

**Issue by informatic-oaib** [globaleaks/GlobaLeaks#3212]
I'm trying to export a report submission that has about 160 attachments (about 2.2MB each). The zip creation starts, but fails to download at about 100-150MB.

**To Reproduce**
1. Go to Reports 
2. Open a report with a lot of attachments (mine is 160 attachments of about 2.2MB each)
3. Click on export
4. The zip creation starts but at about 150-200MB the browser fails to download.
5. Nothing shows on globaleaks.log
6. In access.log, if I download a single file i get this

```
POST /api/token
PUT /api/token/<file-hash>
GET /api/rfile/<hash>?token=<file-hash>
```

But when I try to export the whole report, the access.log only show first two requests and the zip file download fails.

**Expected behavior**
Download the complete report, or throw an error on logs.
As a workaround, let the user select (checkboxes) a number of files (10, 20, etc.), instead of havoing to download one by one.

**Screenshots**
Attached two screenshots of the beggining of the download (Unknown remaining time)
![scr-1](https://user-images.githubusercontent.com/102304607/159932708-97053a48-b276-4738-a383-a6ed958b82c2.png)

and the abnormal termination (Failed)
![scr-2](https://user-images.githubusercontent.com/102304607/159932704-9afa65ef-d92d-4ff6-ae3f-d87e7147ad6d.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: chrome  99.0.4844.74 (64 bits), firefox 98.0.1 (64-bits), also in private mode

**Dedicated Server (only globaleaks installed)
- Ubuntu 20.04.3 LTS (virtualized on VMWare)
- 1x Intel Xeon CPU D-1521
- 2 GB RAM
- 7.4GB of free space in filesystem

**Comment by evilaliv3** [comment]
Thank you for reporting this @informatic-oaib 

We will try a similar condition and get back to you.

**Comment by evilaliv3** [comment]
Hello @informatic-oaib 

I confirm the presence of a bug caused by a large number of files attached on the report due to the way the export module is implement.

Currently in fact it get access first to all the files (opening a file descript for each of them) and only after that it start processing every file one by one.

When more than ~100 files are opened the system depending on the file descriptors limits will trigger an exception for too many open file descriptor, leading to the failure. The failure itself is not problematic because when it happens globaleaks in reaction automatically proceed closing the open files related to the export.

**Comment by evilaliv3** [comment]
Within commit https://github.com/globaleaks/GlobaLeaks/commit/13d9dce59356e46054d031261793a6256e7b70d2 i've changed the implementation geting sure that file descriptors are opened one by one when processing the single files and this would fix this issue.

The patch will be included in the next release 4.9.0 expected to be released in the next days.

Than you again @informatic-oaib for your reporting.
